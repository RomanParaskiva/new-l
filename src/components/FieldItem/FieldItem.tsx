import React from "react";
import classNames from "classnames";

export interface IProps {
    handleClick: () => void
    itemValue: number
}

export const FieldItem = ({
    handleClick,
    itemValue
}: IProps) => {
    const itemStyles = classNames(
        `d-flex w-[15px] h-[15px] border border-solid`,
        itemValue ? "bg-pink-300" : ""
    );

    return <div data-testid="field-item" className={itemStyles} onClick={handleClick} />;
};
