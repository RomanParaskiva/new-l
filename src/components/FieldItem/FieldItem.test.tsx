import { render, screen, act } from '@testing-library/react'
import { fireEvent } from '@testing-library/react'
import { FieldItem } from './FieldItem'

describe('tests FieldItem component', () => {
  test('click on FieldItem component', async () => {
   
      const handleClick = jest.fn();

      render(
        <FieldItem handleClick={handleClick} itemValue={0} />,
      );

      fireEvent.click(screen.getByTestId('field-item'));

      expect(handleClick).toBeCalled();
  })

  test('snapshot FieldItem component', () => {
    act(() => {
      const handleClick = jest.fn()

      const { container } = render(
        <FieldItem handleClick={handleClick} itemValue={0} />,
      )

      expect(container).toMatchSnapshot()
    })
  })
})
