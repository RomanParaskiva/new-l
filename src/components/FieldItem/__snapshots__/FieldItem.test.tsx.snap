// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests FieldItem component snapshot FieldItem component 1`] = `<div />`;
